Action/CheckoutServer/StatusAction.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Payum\Stripe\Action\CheckoutServer;
+
+use Payum\Core\Action\ActionInterface;
+use Payum\Core\Bridge\Spl\ArrayObject;
+use Payum\Core\Exception\RequestNotSupportedException;
+use Payum\Core\Request\GetStatusInterface;
+use Payum\Stripe\Constants;
+use Stripe\PaymentIntent;
+use Stripe\Refund;
+
+class StatusAction implements ActionInterface
+{
+    /**
+     * {@inheritDoc}
+     *
+     * @param GetStatusInterface $request
+     */
+    public function execute($request)
+    {
+        RequestNotSupportedException::assertSupports($this, $request);
+
+        $model = ArrayObject::ensureArrayObject($request->getModel());
+
+        if ($model['error']) {
+            $request->markFailed();
+
+            return;
+        }
+
+        if (false == $model['status']) {
+            $request->markNew();
+
+            return;
+        }
+
+        if ($model['object'] === PaymentIntent::OBJECT_NAME && Constants::STATUS_PROCESSING === $model['status']) {
+            $request->markPending();
+
+            return;
+        }
+
+        if ($model['object'] === Refund::OBJECT_NAME && Constants::STATUS_SUCCEEDED === $model['status']) {
+            $request->markRefunded();
+
+            return;
+        }
+
+        if ($model['object'] === PaymentIntent::OBJECT_NAME && Constants::STATUS_CANCELED == $model['status']) {
+            $request->markCanceled();
+
+            return;
+        }
+
+        if ($model['object'] === PaymentIntent::OBJECT_NAME && Constants::STATUS_SUCCEEDED === $model['status']) {
+            $request->markCaptured();
+
+            return;
+        }
+
+        if ($model['object'] === PaymentIntent::OBJECT_NAME && Constants::STATUS_REQUIRES_PAYMENT_METHOD == $model['status']) {
```

Comment:
if i'm not wrong `Constants::STATUS_REQUIRES_PAYMENT_METHOD == $model['status']` when the authentication failed and customer needs to enter a new card, and `Constants::STATUS_REQUIRES_ACTION == $model['status']` when it's okay but need authentication?

Reply:
Good point, I'm kind torn between `requires_capture` and `requires_confirmation`. It seem that `requires_action` is in-between state and authorized payment intent will have a `requires_capture` state.

> If the selected payment method requires additional authentication steps, the PaymentIntent will transition to the `requires_action` status and suggest additional actions via next_action. If payment fails, the PaymentIntent will transition to the `requires_payment_method` status. If payment succeeds, the PaymentIntent will transition to the `succeeded` status (or `requires_capture`, if capture_method is set to manual).
https://stripe.com/docs/api/payment_intents/confirm